helix-core/src/syntax.rs

```diff
@@ -797,20 +807,20 @@ impl Loader {
             .and_then(|n| n.to_str())
             .and_then(|file_name| self.language_config_ids_by_extension.get(file_name))
```

Comment:
I believe we should remove this special case and handle all existing cases that use this as a glob

Reply:
no those are actual fie extensions. As you can see here this  lookup doesn't look at the file extension but the entires filename instead. So this matches `../rs` but not `../foo.rs`. The file extension check is done later. The check here if for historical reasons to also support specifying full filnames like `Cargo.toml`. This hack has the unfortunate consequence of common file extensions matching the full filename. For example so weird scripting language uses `.res` as a file extension and I commonly use temporary files named just `res`  which  then have the wrong filetype.

So to fix this we should remove this hack (so that a plain string really only matches file extensions) and full filnames should use a glob pattern. So for example `file-types = ["toml", "poetry.lock", "Cargo.lock"]` would become `file-types = ["toml", { glob = "poetry.lock" }, { glob = "Cargo.lock" }]`